src/packet/secret_key_parser.rs

```diff
@@ -38,12 +39,22 @@ fn new_private_key_parser(
     |i: &[u8]| {
         let (i, created_at) = map_opt(be_u32, |v| Utc.timestamp_opt(i64::from(v), 0).single())(i)?;
         let (i, alg) = map(be_u8, PublicKeyAlgorithm::from)(i)?;
-        let (i, params) = parse_pub_priv_fields(alg)(i)?;
+
+        let i = if *key_ver == KeyVersion::V6 {
+            // "scalar octet count for the following public key material" -> skip
+            let (i, _len) = be_u32(i)?;
```

Comment:
shouldn't be skipped, but validated

Reply:
i've taken a stab at using this data, and like the general shape it's taking.

however, i feel like i'm doing error handling wrong, in the nom code: https://github.com/rpgp/rpgp/pull/379/commits/6440eb831ed231ac261fbbca0347fd9678a85f0d#diff-a690e568e7a7967c9ea9d6230ca7798089e653a14d39e45b7cfce5ac262b39eeR21-R30

it looks very clumsy - is there a way to do this more nicely?

Reply:
no, this is unfortunately the way it works atm